### PR TITLE
Feature/fix sending too long

### PR DESCRIPTION
**Pull request by ganfra**
This PR lets sending workers responsible for getting the event in the db instead of passing it through params.
It allows to fix https://github.com/vector-im/element-android/issues/1397

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/store/db/RealmCryptoStore.kt [hunk 1]
```diff
@@ -1191,7 +1191,7 @@ internal class RealmCryptoStore @Inject constructor(
                     .findAll()
                     .mapNotNull { entity ->
                         when (entity.type) {
-                            GossipRequestType.KEY    -> {
+                            GossipRequestType.KEY -> {
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/store/db/RealmCryptoStore.kt:1194]: So you have the same pb than me, Android Studio mess up the alignment of `->`! You pretended it was not the case! Loul!

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/LocalEchoIdentifiers.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+package org.matrix.android.sdk.internal.session.room.send
+
+import com.squareup.moshi.JsonClass
+
+/**
+ * This is used as a holder to pass necessary data to some workers params.
+ */
+@JsonClass(generateAdapter = true)
+data class LocalEchoIdentifiers(val roomId: String, val eventId: String)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/LocalEchoIdentifiers.kt:25]: internal?

**Comment by bmarty** [comment]
Is migration of Work DB safe?

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/LocalEchoIdentifiers.kt (hunk 2 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/LocalEchoIdentifiers.kt]: yup thanks

**Comment by ganfra** [comment]
> Is migration of Work DB safe?

If there is some unsent events they will never sent, but the queue should be free.

**Review by BillCarsonFr — APPROVED** [review]
LGTM

**Pull request merged**